themes/bootstrap3/templates/RecordDriver/EDS/core.phtml

```diff
@@ -7,12 +7,14 @@
     $customLinks = $this->driver->getCustomLinks();
     $accessLevel = $this->driver->getAccessLevel();
     $restrictedView = empty($accessLevel) ? false : true;
+    $coverDetails = $this->record($this->driver)->getCoverDetails('core', 'medium', $thumb);
+    $cover = $coverDetails['html'];
 ?>
 <div class="media" vocab="http://schema.org/" resource="#record" typeof="<?=$this->driver->getSchemaOrgFormats()?> Product">
   <div class="media-left medium img-col">
-    <?php if ($thumb): ?>
-      <img src="<?=$this->escapeHtmlAttr($thumb)?>" class="recordcover" alt="<?=$this->transEsc('Cover Image')?>"/>
-    <?php else: ?>
+    <?php if ($cover): ?>
+      <?=$cover?>
+    <?php else: //TODO an icon as default cover? ?>
```

Comment:
This definitely does not work correctly; the icons never display because getCoverDetails always displays the unavailable image. Also note that there's an EDS/result-list.phtml that also displays icons, but that has not been modified in this PR, and it still seems to work correctly without the ajaxcovers setting, but to never load covers at all when ajaxcovers is enabled.

Would it make sense to provide a mechanism for alternate content to display when a cover is unavailable? This might tie in with the idea of "loading" vs. "unavailable" images -- when we're in AJAX mode, we can display a "loading" image until the AJAX handler is called, then we either replace the loading image with the real content, or else we reveal a hidden element with either the "unavailable" image (in most situations) or the content icon (for EDS).

If that's too complicated, we at least need to figure out a way to disable the AJAX functionality within the context of EDS so that it doesn't break the existing templates.

Any thoughts/preferences?

Reply:
I do agree, that such concept of differantiation loading/unavailable does make sense a could help us to fix the EDS icons/covers... 

I do like the idea, so I'll try to implement it on this PR and will see, how it will go

Reply:
Sounds good! I'll focus on the other open PR's tomorrow to see if I can get them closer to completion, and I'll probably have some time to help with this next week if necessary.

Reply:
@xmorave2, the EDS search results still work correctly with ajax covers disabled, and fail to display any covers (instead showing icons only) with ajax covers enabled.

On the record view, items that have cover images now work correctly with or without ajax covers enabled. However, items that have no cover image are now broken regardless of how the ajax cover setting is configured -- they just show a "Loading..." status that never loads. 

I realize that this is difficult for you to test and develop, and I'm willing to put some time into it... but before I do, I wanted to find out whether you expected this to be working correctly, or if you had any suggestions for which logic I should try to port into the EDS templates.

Reply:
I hope I finally fixed the EDS covers, but I was not able to fully test it. Could you look at it @demiankatz? Thanks!

Reply:
Thanks, I should have time to review later today; I'll let you know how it goes!

Reply:
It took minor adjusting to get it working, but I think it's good now!